app/models/api_key.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class ApiKey < ApplicationRecord
+  include PaperTrailTraceable
+
+  belongs_to :organization
+
+  before_create :set_value
+
+  def generate_value
+    value = SecureRandom.uuid
+    api_key = ApiKey.find_by(value:)
+
+    return generate_value if api_key.present?
+
+    value
+  end
+
+  private
+
+  def set_value
+    self.value = generate_value
+  end
```

Comment:
I would avoid recursion in Ruby when possible and instead of `find_by` use `exists?` for better performance.

also the method could be a private one, is there any reason to expose it a public api of the model?

```suggestion
  before_create :set_key_value
  
  private

  def set_key_value
    begin
      self.value = SecureRandom.uuid 
    end while self.class.exists?(value: self.value)
  end
```

This should do the trick.